UI/Selenium/Steps/HearingListSteps.cs

```diff
@@ -140,5 +146,76 @@ public void ThenCheckTheJudgeReturnsToTheWaitingRoom()
             Driver.FindElement(ParticipantWaitingRoomPage.StartVideoHearingButton).Displayed.Should().BeTrue();
         }
 
+        [When(@"selects the messages tab")]
+        public void WhenSelectsTheMessagesTab()
+        {
+            hearingRoomSteps = new HearingRoomSteps(_scenarioContext);
+            ExtensionMethods.WaitForElementVisible(Driver, ConsultationRoomPage.WaitingRoomIframe);
+            Driver.FindElement(HearingListPage.MessagesTabButton).Click();
+        }
+
+        [When(@"is able to see and send messages to the hearing participants")]
+        public void WhenIsAbleToSeeAndSendMessagesToTheHearingParticipants()
+        {
+            var msg = "Hi Judge, this is a test message";
+            Driver = GetDriver(_hearing.Participant.Where(a => a.Role.Name.ToLower() == "judge").FirstOrDefault().Id, _scenarioContext);
+            _scenarioContext["driver"] = Driver;
+            hearingRoomSteps.WhenTheJudgeCanSendAMessageToAVHOUsingViaHearingRoomChatPanel();
+            Driver = GetDriver(_hearing.Participant.Where(a => a.Role.Name.ToLower() == "vho").FirstOrDefault().Id, _scenarioContext);
+            _scenarioContext["driver"] = Driver;
+            Driver.FindElements(HearingListPage.IMAvailableParticipant)?.FirstOrDefault()?.Click();
+            ExtensionMethods.WaitForElementVisible(Driver, HearingListPage.InstantMessageInput);
+            ExtensionMethods.WaitForElementVisible(Driver, HearingListPage.ChatWindow);
+            Driver.FindElement(HearingListPage.ChatWindow).Text.Should().Contain("Hi, could you please join the hearing");
+            Driver.FindElement(HearingListPage.InstantMessageInput).SendKeys(msg);
+            Driver.FindElement(HearingListPage.InstantMessageButton).Click();
+            ExtensionMethods.WaitForElementVisible(Driver, HearingListPage.DivContainsText(msg));
+            Driver.FindElement(HearingListPage.ChatWindow).Text.Should().Contain(msg);
+            Driver = GetDriver(_hearing.Participant.Where(a => a.Role.Name.ToLower() == "judge").FirstOrDefault().Id, _scenarioContext);
+            _scenarioContext["driver"] = Driver;
+            ExtensionMethods.FindElementWithWait(Driver, HearingRoomPage.JudgeMessageSent(msg), _scenarioContext).Displayed.Should().BeTrue();
+        }
+
+        [Then(@"vho can IM both judge and participants while in the waiting room")]
+        public void ThenVhoCanIMBothJudgeAndParticipants()
+        {
+            var judgeMsg = "Hi Judge, I can see you're in the waiting room now";
+            var participantMsg = "Hello, welcome to the waiting room, enjoy your break";
+            Driver = GetDriver(_hearing.Participant.Where(a => a.Role.Name.ToLower() == "vho").FirstOrDefault().Id, _scenarioContext);
+            _scenarioContext["driver"] = Driver;
```

Comment:
_scenarioContext["driver"] = Driver;
I see the "driver" key in the scenarioContext is being overwritten multiple times in this method. What's the reason behind this?
My suggestion is instead of overwritting the existing key, we can append to Drivers collection. so they can all be disposed properly in AfterScenario